write_test.go

```diff
@@ -0,0 +1,132 @@
+package dicom
+
+import (
+	"testing"
+	"os"
+	"encoding/binary"
+
+  "github.com/stretchr/testify/assert"
+  "github.com/suyashkumar/dicom/pkg/dicomio"
+  "github.com/suyashkumar/dicom/pkg/tag"
+	"github.com/suyashkumar/dicom/pkg/uid"
+)
+
+/*
+FURTHER TESTING
+	- Read written values back in and verify Datsets are the same
+	- With 'wild' DICOMs with high variability, read in, write out, read in, and verify
+*/
+
+// TODO clean this function up big time
+func TestWrite(t *testing.T) {
+	location := "fullwrite.dcm"
+	file, err := os.Create(location)
+	assert.Nil(t, err)
+	defer file.Close()
+
+	mediaStorageSOPClassUID, err := newElement(tag.MediaStorageSOPClassUID, []string{"1.2.840.10008.5.1.4.1.1.1.2"})
+	assert.Nil(t, err)
+	mediaStorageSOPInstanceUID, err := newElement(tag.MediaStorageSOPInstanceUID, []string{"1.2.3.4.5.6.7"})
+	assert.Nil(t, err)
+	transferSyntax, err := newElement(tag.TransferSyntaxUID, []string{uid.ImplicitVRLittleEndian})
+	assert.Nil(t, err)
+	patientName, err := newElement(tag.PatientName, []string{"Robin Banks"})
+	assert.Nil(t, err)
+
+	elems := []*Element{
+							 mediaStorageSOPClassUID,
+							 mediaStorageSOPInstanceUID,
+							 transferSyntax,
+							 patientName,
+	}
+
+	ds := &Dataset{Elements: elems}
+
+	err =  Write(file, ds, func (set *writeOptSet){})
+	assert.Nil(t, err)
+
+	// TODO verify that the correct values are written
+}
+
+// TODO clean this function up big time
+func TestWriteFileHeader(t *testing.T) {
+	location := "fileheader.dcm"
+	file, err := os.Create(location)
+	assert.Nil(t, err)
+	defer file.Close()
+
+  w := dicomio.NewWriter(file, binary.LittleEndian, false)
+
+	mediaStorageSOPClassUID, err := newElement(tag.MediaStorageSOPClassUID, []string{"1.2.840.10008.5.1.4.1.1.1.2"})
+	assert.Nil(t, err)
+	mediaStorageSOPInstanceUID, err := newElement(tag.MediaStorageSOPInstanceUID, []string{"1.2.3.4.5.6.7"})
+	assert.Nil(t, err)
+	transferSyntax, err := newElement(tag.TransferSyntaxUID, []string{uid.ImplicitVRLittleEndian})
+	assert.Nil(t, err)
+	metaElems := []*Element{
+							 mediaStorageSOPClassUID,
+							 mediaStorageSOPInstanceUID,
+							 transferSyntax,
+	}
+	ds := &Dataset{Elements: metaElems}
+
+	err = writeFileHeader(w, ds, metaElems, func (set *writeOptSet){})
+	assert.Nil(t, err)
+
+	// TODO Verify the the corrrect things were written to the file header
+}
+
+func TestEncodeElementHeader(t *testing.T) {}
+
+func TestWriteValue(t *testing.T) {}
+
+func TestWriteTag(t *testing.T) {}
+
+func TestWriteVRVL(t *testing.T) {}
+
+func TestVerifyVR(t *testing.T) {
```

Comment:
we could use the test cases pattern here, but this is more of a note to myself to revisit this (we can do it in a future PR too)
e.g. 

```go
cases := []struct{
    name string
    tag tag.Tag
    vr string
} {
  // test cases
}

for _, tc := range cases {
    t.Run(tc.name, func(t *testing.T) {
        // test case
    })
}
```